kratos/utilities/volume_inside_voxel_utility.cpp

```diff
@@ -0,0 +1,393 @@
+            double l = Distance(ends[0], ends[1]);
+            Length[i] = l;
+        }
+
+        for (int i = 0; i < rTriangles.size(); i++) {
+            int Result = 0; 
+            array_1d<double,3> Intersection;
+            int j = 0;
+
+            /*We will iterate through the edges using a while loop, so that if a triangles intersects 2 edges (unlikely 
+            but possible), only one will be taken into account to create the matrixes. */
+            while(!Result && j < Edges.size()) { 
+                PointsArrayType ends = Edges[j].Points();
+                Result = IntersectionUtilities::ComputeTriangleLineIntersection(rTriangles[i],ends[0],ends[1],Intersection);
+
+                if (Result) {
+                    double Dist = Distance(ends[0], Intersection);
+                    if ( Dist < (MinDistanceToNode[j].first*Length[j])) {
+                        MinDistanceToNode[j].first = Dist/Length[j];
+                    } 
+
+                    double Dist2 = Distance(ends[1], Intersection);
+                    if (Dist2 < (MinDistanceToNode[j].second*Length[j])) {
+                        MinDistanceToNode[j].second = Dist2/Length[j];
+                    } 
+                }
+                j++;
+            }
+        }
+
+        std::vector<std::vector<double>> Neighbours{{3,1},{0,2},{1,3},{0,2}};  
+        for(int i = 0; i < Nodes.size(); i++ ) {
+            double Factor = GetFactor(Nodes, Neighbours,i);
+            double PartialArea;
+            if (Nodes[i].GetSolutionStepValue(DISTANCE) > 0) {
+                PartialArea = Factor*MinDistanceToNode[(i+3)%4].second*MinDistanceToNode[i].first;
+            } else  {
+                PartialArea = 1.0/Nodes.size() - Factor*MinDistanceToNode[(i+3)%4].second*MinDistanceToNode[i].first;
+            }
+            Area += PartialArea;
+        }
+        return Area;    
+    }
+
+    /***********************************************************************************
+     **********************************************************************************/
+
+    double VolumeInsideVoxelUtility::HexaVolume2D(
+        const GeometryType& rFace,  
+        const GeometryArrayType& rTriangles     
+    ) {
+        double Area = 0;
+        GeometryArrayType Edges = rFace.GenerateEdges();
+        PointsArrayType Nodes = rFace.Points(); 
+        double FaceArea = TetraVolume(Nodes);
+        std::vector<std::pair<double,double>> MinDistanceToNode(Edges.size(),{1,1}); 
+        
+        int NodesInside = 0;
+        for (int i = 0; i < Nodes.size(); i++) {
+            if (Nodes[i].GetSolutionStepValue(DISTANCE) > 0) NodesInside++;
+        }
+
+        if(NodesInside == 3) {
+            for (int i = 0; i < Nodes.size(); i++)  {
+                Nodes[i].GetSolutionStepValue(DISTANCE) = (-1)*Nodes[i].GetSolutionStepValue(DISTANCE);
+            }
+            double Area = HexaVolume2D(rFace,rTriangles);
```

Comment:
```suggestion
            const double area = HexaVolume2D(rFace,rTriangles);
```
snake_case when it is not an input. And use const if possible